internal/util/validation/validation.go

```diff
@@ -107,6 +113,68 @@ func ValidateBearerToken(token *string, path string) []error {
 	return asErrors(errs)
 }
 
+func ValidateCSRUsages(u *[]string) []error {
+	errs := field.ErrorList{}
+	requiredAllOf := []string{"client auth", "CA:false"}
+	notAllowed := []string{}
+
+	for _, usage := range *u {
+		for _, na := range notAllowed {
+			if usage == na {
+				err := fmt.Sprintf("usage not allowed: %s\n", na)
+				errs = append(errs, field.Invalid(fieldPathFor("spec.usages"), u, err))
+			}
+		}
+		for i, r := range requiredAllOf {
+			if usage == r {
+				// remove fulfilled req from list
+				requiredAllOf = append(requiredAllOf[:i], requiredAllOf[i+1:]...)
+			}
+		}
+	}
+	if len(requiredAllOf) > 0 {
```

Comment:
NIT: you can consider finding the unfulfilled requirements by:
`lo.Without(requiredAllOf, (*u)...)`

Reply:
I ended up making the slices into maps (see comments on #453 ) but this is a good library to know about, thanks.